rest-client-base/src/main/java/com/wultra/core/rest/client/base/DefaultRestClient.java

```diff
@@ -110,13 +116,24 @@ private void initializeWebClient() throws RestClientException {
         final SslContext sslContext = SslUtils.prepareSslContext(config);
         HttpClient httpClient = HttpClient.create();
         if (sslContext != null) {
-            httpClient = httpClient.secure(sslContextSpec -> sslContextSpec.sslContext(sslContext));
+            httpClient = httpClient.secure(sslContextSpec -> {
+                final SslProvider.Builder sslProviderBuilder = sslContextSpec.sslContext(sslContext);
+                config.getHandshakeTimeout().ifPresent(it -> {
+                    logger.debug("Setting handshake timeout {}", it);
+                    sslProviderBuilder.handshakeTimeout(it);
```

Comment:
Have we tested this configuration? I am a bit concerned that we are using a builder and the `build()` method is not called. Perhaps it works, but I would just like to doublecheck this.

Reply:
I will add a test, if you are ok with this suggestion.

Reply:
Build method is called in
```java
	public final HttpClient secure(Consumer<? super SslProvider.SslContextSpec> sslProviderBuilder) {
		Objects.requireNonNull(sslProviderBuilder, "sslProviderBuilder");
		SslProvider.SslContextSpec builder = SslProvider.builder();
		sslProviderBuilder.accept(builder);
		SslProvider sslProvider = HttpClientSecure.sslProvider(((SslProvider.Builder) builder).build());
		if (sslProvider.equals(configuration().sslProvider)) {
			return this;
		}
		HttpClient dup = duplicate();
		dup.configuration().sslProvider = sslProvider;
		return dup;
	}
```

Reply:
Test added.